aws/resource_aws_ecr_repository.go

```diff
@@ -55,30 +56,29 @@ func resourceAwsEcrRepositoryCreate(d *schema.ResourceData, meta interface{}) er
 		RepositoryName: aws.String(d.Get("name").(string)),
 	}
 
-	log.Printf("[DEBUG] Creating ECR resository: %s", input)
+	log.Printf("[DEBUG] Creating ECR repository: %#v", input)
 	out, err := conn.CreateRepository(&input)
 	if err != nil {
-		return err
+		return fmt.Errorf("error creating ECR repository: %s", err)
 	}
 
 	repository := *out.Repository
 
 	log.Printf("[DEBUG] ECR repository created: %q", *repository.RepositoryArn)
 
-	d.SetId(*repository.RepositoryName)
+	d.SetId(aws.StringValue(repository.RepositoryName))
+	// ARN required for setting any tags.
 	d.Set("arn", repository.RepositoryArn)
-	d.Set("registry_id", repository.RegistryId)
-
-	return resourceAwsEcrRepositoryRead(d, meta)
+	return resourceAwsEcrRepositoryUpdate(d, meta)
```

Comment:
`Update` after `Create` usually winds up causing longterm maintenance issues with duplicate API calls, etc. due to lack of `d.IsNewResource()` conditionals. Let's just copy the `setTagsECR()` bits into `Create` to prevent that problem in the future. 👍 (At some point we will likely have a provider linting tool that errors when calling `Update` after `Create`.)

Reply:
👍